### PR TITLE
 Service/FRD: Stub HasLoggedIn, IsOnline, Login and Logout

### PR DESCRIPTION
**Pull request by valentinvanelslande**
I do not know if the errors are the right ones...

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3409)

### CODE REVIEW

File: src/core/hle/service/frd/frd_u.cpp [hunk 1]
```diff
@@ -9,10 +9,10 @@ namespace Service {
 namespace FRD {
 
 const Interface::FunctionInfo FunctionTable[] = {
-    {0x00010000, nullptr, "HasLoggedIn"},
-    {0x00020000, nullptr, "IsOnline"},
-    {0x00030000, nullptr, "Login"},
-    {0x00040000, nullptr, "Logout"},
+    {0x00010000, HasLoggedIn, "HasLoggedIn"},
+    {0x00020000, IsOnline, "IsOnline"},
+    {0x00030000, Login, "Login"},
```
Comment (mailwl) [inline src/core/hle/service/frd/frd_u.cpp]: change please code to 0x00030002, here is error

File: src/core/hle/service/frd/frd.cpp [hunk 2]
```diff
@@ -89,6 +90,51 @@ void GetFriendAttributeFlags(Service::Interface* self) {
...
+void IsOnline(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+    cmd_buff[2] = 0;                  // Not online
+    LOG_WARNING(Service_FRD, "(STUBBED) called");
+}
+
+void Login(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
```
Comment (mailwl) [inline src/core/hle/service/frd/frd.cpp]: according https://www.3dbrew.org/wiki/FRDU:Login, second parameter is completion event handler

Comment (valentinvanelslande) [reply src/core/hle/service/frd/frd.cpp]: How i can get that from cmd_buff?

Comment (mailwl) [reply src/core/hle/service/frd/frd.cpp]: for example, https://github.com/mailwl/citra/blob/friends/src/core/hle/service/frd/frd.cpp#L46
but, this method is obsoleted, i am not sure it has to be used for new stubbing...

**Comment by valentinvanelslande** [comment]
@wwylele what error i can use?

**Comment by mailwl** [comment]
i saw the game which uses this functions, but don't remember the name (AFAIK, it don't crash anyway, just red lines in log)

**Comment by valentinvanelslande** [comment]
test for checking if returns error:
https://gist.github.com/valentinvanelslande/c2b165cef72eded8733b9006d4fbee72
build: [.3dsx](https://github.com/citra-emu/citra/files/1670497/frd_test.zip)

**Pull request closed without merging**